Add metric for number of offloaded timelines

arpad-m opened the pull request:
We want to keep track of the number of offloaded timelines. It's a per-tenant shard metric because each shard makes offloading decisions on its own.

erikgrinaker reviewed (approved):
This seems a bit out of place in the manifest upload, but I suppose it's pragmatic.

arpad-m commented:
Added a test to read the metric. Also fixed a deadlock (it tried to lock the offloaded timelines accessor object while already having a valid lock accessor).
